pkg/machines/components/vm/bootOrderModal.jsx

```diff
@@ -0,0 +1,301 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import cockpit from 'cockpit';
+import {
+    Button,
+    ButtonGroup,
+    Icon,
+    ListView,
+    ListViewItem,
+    Modal
+} from 'patternfly-react';
+
+import { ModalError } from './../notification/inlineNotification.jsx';
+import {
+    findHostNodeDevice,
+    getSortedBootOrderDevices,
+    rephraseUI,
+    vmId
+} from '../../helpers.js';
+import {
+    changeBootOrder,
+    getVm
+} from '../../actions/provider-actions.js';
+
+import './bootOrderModal.css';
+
+const _ = cockpit.gettext;
+const UP = -1;
+const DOWN = 1;
+
+function getBootOrderDevices(vm) {
+    let devices = getSortedBootOrderDevices(vm.inactiveXML);
+
+    devices.forEach(dev => {
+        dev.checked = typeof dev.bootOrder !== 'undefined';
+        dev.initialOrder = parseInt(dev.bootOrder);
+    });
+
+    return devices;
+}
+
+const DeviceInfo = ({ descrClass, descr, valueClass, value }) => {
+    return (
+        <div>
+            <span className={descrClass}>
+                <label className='control-label'>
+                    {descr}
+                </label>
+            </span>
+            <span className={valueClass}>
+                {value}
+            </span>
+        </div>
+    );
+};
+
+const DeviceRow = ({ idPrefix, device, index, onToggle, upDisabled, downDisabled, moveUp, moveDown, nodeDevices }) => {
+    let heading;
+    let additionalInfo = [];
+
+    const addOptional = (additionalInfo, value, descr) => {
+        if (value) {
+            additionalInfo.push(<DeviceInfo
+                                    descrClass='boot-order-source-descr' descr={descr}
+                                    valueClass='boot-order-source-value' value={value}
+                                    key={index + descr} />);
+        }
+    };
+
+    if (device.type === "disk") {
```

Comment:
I think switch case is better for this else if hell, but I am not strongly opinionated on this.

Reply:
Yeah, with switch it looks cleaner. I also changed if-else statements for switch in helper and libvirt-common.